### ISSUE TITLE
inconsisten documentation placement rules?

### ISSUE THREAD

**Issue by EinScott** [beefytech/Beef#1196]
i was trying to make some documentation for the ide's way to display documentation. Though sometimes it seems that documentation is expected to be placed ABOVE attributes on the target, sometimes (private members?) it needs to be below.

```
static
	{
		public static void DoTestDoc()
		{
			DoAThing();

			
			AA<int> d;
			d.DoAThingP();
			d.[Friend]DoAThing();

			DoAnotherThing(0, 0);
		}
		
		[Optimize]
		/// Nothing is allowed to be between documentation and target.
		/// Using multiple lines like this is also fine.
		public static void DoAThing() {}

		/// Documentation also works for types
		struct AA<T>
		{
			[Optimize]
			/// Same here
			public void DoAThingP() {}


			/// Here it ONLY works the other way around??
			[Optimize]
			void DoAThing() {}
		}

		/**
		* If you have a really long explainer here, you may not actually want to show that in the autcompletion.
		* @brief Allows you to select only this line to be shown.
		* 
		* @param a This is shown when writing a call to this function and placing parameter "a".
		* @param b After placing the ',' after the first arg, the documentation for b/the second argument will show up instead.
		*/
		public static void DoAnotherThing(int a, int b) {}
	}
```

**Comment by bfiete** [comment]
Documentation should go before the attributes. Parsing fixed at 7436b201be98ad8f07d90b9cb1d169226798de6f

**Comment by EinScott** [comment]
I see, thanks!
